File: app/src/main/java/me/kirimin/mitsumine/MyApplication.java
package me.kirimin.mitsumine;

import com.activeandroid.ActiveAndroid;
import com.activeandroid.Configuration;

import android.app.Application;

public class MyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Configuration.Builder builder = new Configuration.Builder(getBaseContext());
        builder.setCacheSize(1024 * 1024 * 4);
        builder.setDatabaseName("mitsumine.db");
        builder.setDatabaseVersion(5);
        ActiveAndroid.initialize(builder.create(), true);
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
        ActiveAndroid.dispose();
    }
}

File: app/src/main/java/me/kirimin/mitsumine/MyApplication.kt
package me.kirimin.mitsumine

import com.activeandroid.ActiveAndroid
import com.activeandroid.Configuration

import android.app.Application
import me.kirimin.mitsumine._common.di.component.AppComponent
import me.kirimin.mitsumine._common.di.component.DaggerAppComponent
import me.kirimin.mitsumine._common.di.module.AppModule
import me.kirimin.mitsumine._common.di.module.InfraModule
import me.kirimin.mitsumine._common.di.module.ViewModule


class MyApplication : Application() {

    private lateinit var applicationComponent: AppComponent

    override fun onCreate() {
        super.onCreate()
        // Dagger
        initializeInjector()
        // AA
        val builder = Configuration.Builder(baseContext)
        builder.setCacheSize(1024 * 1024 * 4)
        builder.setDatabaseName("mitsumine.db")
        builder.setDatabaseVersion(6)
        ActiveAndroid.initialize(builder.create(), true)
    }

    override fun onTerminate() {
        super.onTerminate()
        ActiveAndroid.dispose()
    }

    private fun initializeInjector() {
        applicationComponent = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .build()
    }

    fun getApplicationComponent(): AppComponent {
        return applicationComponent
    }
}
